crates/codegen/language/definition/src/model/manifest/section.rs

```diff
@@ -0,0 +1,12 @@
+use crate::model::Topic;
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[codegen_language_internal_macros::derive_spanned_type(
+    codegen_language_internal_macros::ParseInputTokens,
+    codegen_language_internal_macros::WriteOutputTokens
+)]
+pub struct Section {
```

Comment:
Agree with you on `manifest.rs`. Moved that. Although for others, I suggest keeping them split:

- Because it is also about hierarchy. `fragment.rs` is used by all mods in `terminals/*`, similar thing for `field.rs` and `non-terminals/*`. I think indicating this relationship is more important than their size.
- Stability: while some of these are still just data types, we are already adding all sorts of APIs/traits/utils for codegen, and I don't want to keep moving the types around as they grow. I already have a few drafts ready to go after this one.

Reply:
> I already have a few drafts ready to go after this one.

This makes sense and I don't want to block you; it's also hard to tell what you specifically might want and what not, but defaulting all of these to start with a new file rather than starting with a more balanced approach seems a bit extreme (i.e. `manifest.rs` that was immediately combined now).

> Because it is also about hierarchy.

I fully agree and what I'm proposing doesn't eliminate that.

For instance, defining `Scanner`, `Token{Item,Definition}` and `FragmentItem` already in `terminals/mod.rs`  is not really unexpected; scanners and tokens are *the* basis for terminals (as a concept) and FragmentItem as data is never used directly and only by-reference in Scanner (which is already used by the two in FragmentItem, TokenDefinition but also in TriviaItem, however that should probably be separated since trivia is kind-of-token-kind-of-rule anyway).

This leaves us with 3 related concepts defined in one place, at the `mod terminals` (IMO separating them makes more harm than good in this case), with more specialized items like keywords and trivia in their own submodule and I think that this indicates the relationship clearly, by grouping together relevant concepts.

However, that's not to say that I don't see the value in symmetry of defining `XXXItem` in the same, uniform way. I just wish we had less 15-20 line length files exporting a single data type.